vars/buildPlugin.groovy

```diff
@@ -152,29 +136,39 @@ def call(Map params = [:]) {
                         }
 
                         stage("Archive (${stageIdentifier})") {
-                            if (isMaven && archiveFindbugs) {
-                                def fp = [pattern: params?.findbugs?.pattern ?: '**/target/findbugsXml.xml']
-                                if (params?.findbugs?.unstableNewAll) {
-                                    fp['unstableNewAll'] = "${params.findbugs.unstableNewAll}"
-                                }
-                                if (params?.findbugs?.unstableTotalAll) {
-                                    fp['unstableTotalAll'] = "${params.findbugs.unstableTotalAll}"
-                                }
-                                findbugs(fp)
-                            }
-                            if (isMaven && archiveCheckstyle) {
-                                def cp = [pattern: params?.checkstyle?.pattern ?: '**/target/checkstyle-result.xml']
-                                if (params?.checkstyle?.unstableNewAll) {
-                                    cp['unstableNewAll'] = "${params.checkstyle.unstableNewAll}"
-                                }
-                                if (params?.checkstyle?.unstableTotalAll) {
-                                    cp['unstableTotalAll'] = "${params.checkstyle.unstableTotalAll}"
-                                }
-                                checkstyle(cp)
-                            }
                             if (failFast && currentBuild.result == 'UNSTABLE') {
                                 error 'There were test failures; halting early'
                             }
+
+                            if (first) {
+                                referenceJobName = env.JOB_NAME.substring(0, env.JOB_NAME.lastIndexOf("/") + 1) + "master"
+                                echo "Recording static analysis results on '${stageIdentifier}' using reference job '${referenceJobName}'"
+
+                                recordIssues enabledForFailure: true,
+                                        tool: mavenConsole(),
+                                        referenceJobName: referenceJobName
+                                recordIssues enabledForFailure: true,
+                                        tools: [java(), javaDoc()],
+                                        sourceCodeEncoding: 'UTF-8',
+                                        filters:[excludeFile('.*Assert.java')],
+                                        referenceJobName: referenceJobName
+                                recordIssues tools: [spotBugs(pattern: '**/target/spotbugsXml.xml'),
```

Comment:
It would make sense to read `findbugsXml` as well

Reply:
I think a quality gate needs to be defined for spotbugs so that the build gets marked as unstable, currently it's just green which isn't good.

Reply:
Yes, I can add quality gates. But which ones should we use? How many new, how many total warnings?

Reply:
0 to both and make it configurable so it can be overridden?

Reply:
After thinking a little bit about the old style with a property I came to another idea: it might be more powerful if we let the developers decide on how to configure the whole step: 

https://github.com/jenkins-infra/pipeline-library/pull/121/commits/e4d86e8dcd43cc4e213a62b6af0696dbcd5080c8

What do you think about such an option? Then we can have meaningful defaults but the developers can override everything as they like.  Otherwise we would have a lot of duplicate properties that are part of the `buildPlugin` code and my plugin. 

If this is ok for you then I also add such a `map` for the other tools (checkstyle, etc.).

Reply:
Great idea and works for me

Reply:
Tested https://ci.jenkins.io/job/Plugins/job/slack-plugin/job/master/ and https://ci.jenkins.io/job/Plugins/job/slack-plugin/view/change-requests/job/PR-707/

Works 👍